core/backend/src/IModelDb.ts

```diff
@@ -1403,34 +1404,51 @@ export namespace IModelDb { // eslint-disable-line no-redeclare
      * @throws [[IModelError]] if unable to insert the model.
      */
     public insertModel(props: ModelProps): Id64String {
-      if (props.isPrivate === undefined) // temporarily work around bug in addon
-        props.isPrivate = false;
+      const json = props instanceof Model ? props.toJSON() : props;
+      if (json.isPrivate === undefined) // temporarily work around bug in addon

```

Comment:
@scsewall added that. His commit comments don't indicate what the actual bug was. Nor does his [PR](https://dev.azure.com/bentleycs/iModelTechnologies/_git/692a403b-7ecc-4797-adb1-e6f5c7080aa9/pullrequest/11777).
```
commit ffb57f16b95a6eed7738d8e91a32dcd517a5f660
Author: Shaun Sewall <36768600+scsewall@users.noreply.github.com>
Date:   Tue Dec 4 23:14:48 2018 +0000

    Merged PR 11777: Temporarily work around Model.isPrivate bug in addon

    Temporarily work around Model.isPrivate bug in addon

diff --git a/common/changes/@bentley/imodeljs-backend/insertModel-isPrivate-workaround_2018-12-04-23-00.json b/common/changes/@bentley/imodeljs-backend/insertModel-isPrivate-workaround_2018-12-04-23-00.json
new file mode 100644
index 0000000000..ade67768bc
--- /dev/null
+++ b/common/changes/@bentley/imodeljs-backend/insertModel-isPrivate-workaround_2018-12-04-23-00.json
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "",
+      "packageName": "@bentley/imodeljs-backend",
+      "type": "none"
+    }
+  ],
+  "packageName": "@bentley/imodeljs-backend",
+  "email": "shaun.sewall@bentley.com"
+}
\ No newline at end of file
diff --git a/core/backend/src/IModelDb.ts b/core/backend/src/IModelDb.ts
index 22a35f9a94..2a21a870fe 100644
--- a/core/backend/src/IModelDb.ts
+++ b/core/backend/src/IModelDb.ts
@@ -966,6 +966,9 @@ export namespace IModelDb {
      * @throws [[IModelError]] if unable to insert the model.
      */
     public insertModel(props: ModelProps): Id64String {
+      if (props.isPrivate === undefined) // temporarily work around bug in addon
+        props.isPrivate = false;
+
       const jsClass = this._iModel.getJsClass<typeof Model>(props.classFullName);
       if (IModelStatus.Success !== jsClass.onInsert(props))
         return Id64.invalid;
```

Reply:
@scsewall can you tell us what this bug was and if it still exists?

Reply:
The m_isPrivate variable on the native code side was not being initialized and it was only getting set if the props contained a value for isPrivate (undefined would let it remain uninitialized). The real fix is on the native code side but I made a fix in TypeScript also so that models would not mysteriously/randomly turn private. Now that the native code fix is in place, we can remove the TypeScript workaround.

Reply:
Bug no longer exists.